lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -0,0 +1,54 @@
+require 'flay'
+require 'json'
+require 'cc/engine/analyzers/helpers/main'
+require 'pry'
+
+module CC
+  module Engine
+    module Analyzers
+      module Ruby
+        class Main
+          include ::CC::Engine::Analyzers::Helpers
+
+          def initialize(directory:, engine_config:, io:)
+            @directory = directory
+            @engine_config = engine_config || {}
+            @io = io
+          end
+
+          def run
+            flay.process(*analyzed_files)
+
+            flay.report(StringIO.new).each do |issue|
+              all_locations = issue.locations
+
+              all_locations.each do |location|
+                other_locations = find_other_locations(all_locations, location)
+                io.puts "#{new_violation(issue, location, other_locations).to_json}\0"
+              end
+            end
+          end
+
+          private
+
+          attr_reader :directory, :engine_config, :io
+
+          def filter_files(files)
+            return files if engine_config["exclude_paths"].nil?
+            files - excluded_files
+          end
+
+          def mass_threshold
+            mass = engine_config.fetch('config', {}).fetch('ruby', {}).fetch('mass_threshold', {})
```

Comment:
This seems to default to `{}` them set as `10` if `empty?`. Would it make more sense to use `fetch("mass_threshold", 10)`?